src/main/java/com/google/sps/servlets/TasksServlet.java

```diff
@@ -52,6 +61,70 @@ public TasksServlet(
     this.tasksClientFactory = tasksClientFactory;
   }
 
+  private List<Task> getTasks(TasksClient tasksClient) throws IOException {
+    List<TaskList> taskLists = tasksClient.listTaskLists();
+    List<Task> tasks = new ArrayList<>();
+    for (TaskList taskList : taskLists) {
+      tasks.addAll(tasksClient.listTasks(taskList));
+    }
+    return tasks;
+  }
+
+  private List<String> getTaskListTitles(List<TaskList> taskLists) throws IOException {
+    return taskLists.stream().map(taskList -> taskList.getTitle()).collect(Collectors.toList());
+  }
+
+  private int getTasksToComplete(List<Task> tasks) {
+    List<Task> tasksCompletedToday =
+        tasks.stream().filter(task -> task.getHidden() == null).collect(Collectors.toList());
+    return tasksCompletedToday.size();
+  }
+
+  private int getTasksDueToday(List<Task> tasks) {
+    String today = LocalDate.now().toString();
+    List<Task> tasksDueToday =
+        tasks.stream()
+            .filter(task -> task.getDue() != null && task.getDue().contains(today))
+            .collect(Collectors.toList());
+    return tasksDueToday.size();
+  }
+
+  private int getTasksCompletedToday(List<Task> tasks) {
+    ZoneId zoneId = ZoneId.systemDefault();
+    Instant startOfDay = LocalDate.now(zoneId).atStartOfDay(zoneId).toInstant();
+    Instant endOfDay = LocalDate.now(zoneId).plusDays(1).atStartOfDay(zoneId).toInstant();
+    List<Task> tasksCompletedToday = new ArrayList<>();
+    for (Task task : tasks) {
+      // getHidden is defined for completed tasks
+      if (task.getHidden() != null) {
+        Instant completionDateTime = ZonedDateTime.parse(task.getUpdated()).toInstant();
+        if (completionDateTime.isAfter(startOfDay) && completionDateTime.isBefore(endOfDay)) {
+          tasksCompletedToday.add(task);
+        }
+      }
+    }
+    return tasksCompletedToday.size();
+  }
+
+  private int getTasksOverdue(List<Task> tasks) {
+    ZoneId zoneId = ZoneId.systemDefault();
+    String zoneOffset = zoneId.getRules().getOffset(Instant.now()).toString();
+    Instant endOfDay = LocalDate.now(zoneId).plusDays(1).atStartOfDay(zoneId).toInstant();
+    List<Task> tasksOverdue = new ArrayList<>();
+    for (Task task : tasks) {
+      if (task.getDue() != null) {
+        // Correct default timezone UTC to system's timezone
```

Comment:
This is good for now, but the system's timezone may not be the same as the user's timezone. We maybe should chat offline about how to handle timezones and get this working on the server once the tasks servlets are finalized a bit more

Reply:
Agreed!

Reply:
`RESOLVED`